### PR TITLE
Soback 67 gps location search

### PR DESCRIPTION
**Pull request by jesperlauridsen**
GPS location search branch

Location search added
split view after searching is done
Images bundled when serveral are close untill you zoom closer
Button added to the frontpage
Minor style adjustments to the frontpage

@thomasegense addeded for functionality check
@jorntx added for code review.

### CODE REVIEW

**Review by jorntx — APPROVED** [review]
I really like how you build the UI for this function - so neat with the screen splitter. I couldn't get the images to work (tomcat returns 400) but I think it is an index problem and the the JS code.

Great reuse of already existing functionality 👍 

Code wise there wasn't much so I will approve tight away with the following remarks:

1. There was some formatting that could be better
2. In a future version you could consider splitting some of the core Leaflet logic out sp the GpsSearch component would be more clean.

UI:
- It is a shame that you cannot trigger a search for image og search for links to image without loosing your map search state. I tried both ordinary click and open in new tab click. Bot ways result in loss of my map and results.

File: src/js/src/components/SearchBox.vue [hunk 1]
```diff
@@ -51,11 +51,17 @@
         </div>
       </div>
       <div class="tools">
-        <span @click="showUploadFileSearch = !showUploadFileSearch">Search with uploaded file</span>
+        <button class="searchByFileButton" @click="showUploadFileSearch = !showUploadFileSearch">
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/components/SearchBox.vue:54]: Good catch.

File: src/js/src/components/modalComponents/GpsSearch.vue [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+      })
+      resizeObserver.observe(document.getElementById('gpsMap'))
+      
+      this.selected = L.featureGroup().addTo(this.searchMap)
+      this.searchMap.on('click', this.setNewSearchArea)
+    },
+    doGeoSearch() {
+      this.setLoadingStatus(true)
+      requestService.fireGeoImageSearchRequest(this.imgQuery,this.latitude,this.longitude,this.radius)
+      .then(result => (this.imgResults = result.response,this.mapSize = 'half', this.resultSize = 'half', this.plotImagesOnMap(this.imgResults.images), this.setLoadingStatus(false)), error => (console.log('Error in seaching for images by location.'), this.setLoadingStatus(false)))
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/components/modalComponents/GpsSearch.vue:121]: Urgh - will format. For you!

File: src/js/src/components/modalComponents/GpsSearch.vue [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+      this.imageLayer = L.markerClusterGroup()
+      images.forEach((item, index) => {
+        let newMarker = L.marker(new L.LatLng(item.latitude,item.longitude), { title:item.urlNorm } )
+        newMarker.bindPopup(`<span title="${item.resourceName}">${item.resourceName}</span><img src="${item.imageUrl}&height=200&width=200" />`)
+        this.imageLayer.addLayer(newMarker)
+      })	
+      this.searchMap.addLayer(this.imageLayer)
+    },
+    dividerPosition() {
+      let decision = ''
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/components/modalComponents/GpsSearch.vue:155]: :D

File: src/js/src/components/modalComponents/GpsSearch.vue [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+      }
+      else {
+        switch(this.mapSize) {
+          case 'full':   this.mapSize = 'half'
+                         this.resultSize = 'half'
+                         break
+          case 'half' :  this.mapSize = 'hidden'
+                         this.resultSize = 'full'
+                         break
+        }
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/components/modalComponents/GpsSearch.vue:190]: :+1:

File: src/js/src/components/searchResults/ImageSearchResults.vue [hunk 5]
```diff
@@ -1,25 +1,25 @@
 <template>
   <div>
-    <span>Showing <span class="highlightText">{{ results.images.length }}</span> images matching query. </span>
+    <span>Showing <span class="highlightText">{{ imgResults.images.length }}</span> images matching query. </span>
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/components/searchResults/ImageSearchResults.vue:3]: :+1:

**Review by jorntx — CHANGES_REQUESTED** [review]
Corrections and additions look good but now I can't start the GPS Search. I get the following error:

Screenshot 2020-11-03 at 07 27 32

**Review by thomasegense — APPROVED** [review]
Approved. Tested branch, did not read code.

**Review by jorntx — APPROVED** [review]
Works now! Let's get this show on the road.

**Comment by jorntx** [comment]
Please resolve conflicts @jesperlauridsen  - thanx.

**Pull request merged**